math/mathcore/inc/TMath.h

```diff
@@ -1002,6 +1006,50 @@ Iterator TMath::LocMin(Iterator first, Iterator last) {
    return std::min_element(first, last);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Returns the one-dimensional gradient of a vector with length n. The default step size h is 1.
+
```

Comment:
returns nullptr if n<2 or f==nullptr

Reply:
specify also in docu what method is used (central, forward, etc.)

And that the returned result is owned by caller and must thus be deleted later

Specify size of result

Reply:
specify that lead to undefined behavior if n does not match the length of f